@here/harp-materials/lib/CirclePointsMaterial.ts

```diff
@@ -43,46 +45,47 @@ export interface CirclePointsMaterialParameters extends THREE.ShaderMaterialPara
      * Point size.
      */
     size?: number;
-}
 
-const DEFAULT_CIRCLE_SIZE = 1;
+    /**
+     * Point color.
+     */
+    color?: THREE.Color;
+}
 
 /**
  * Material designed to render circle points. Note that it is always transparent since the circle
  * shape is created with an alpha channel to benefit an antialising that a mere `discard` could
  * not bring.
  */
 export class CirclePointsMaterial extends THREE.ShaderMaterial {
-    isCirclePointsMaterial: true;
-    uniforms: { [uniform: string]: THREE.IUniform };
-    vertexShader: string;
-    fragmentShader: string;
+    static readonly DEFAULT_CIRCLE_SIZE = 1;
 
-    private m_size: number;
     private m_color: THREE.Color;
+    private m_opacity: number;
 
     /**
      * Constructs a new `CirclePointsMaterial`.
      *
      * @param parameters The constructor's parameters.
      */
     constructor(parameters: CirclePointsMaterialParameters = {}) {
-        parameters.depthTest = false;
+        const { size, color, opacity, ...shaderParams } = parameters;
+        shaderParams.name = "CirclePointsMaterial";
+        shaderParams.vertexShader = vertexShader;
+        shaderParams.fragmentShader = fragmentShader;
+        shaderParams.depthTest = false;
 
-        super(parameters);
+        super(shaderParams);
         enforceBlending(this);
 
-        this.isCirclePointsMaterial = true;
         this.type = "CirclePointsMaterial";
-        this.vertexShader = vertexShader;
-        this.fragmentShader = fragmentShader;
-
-        this.m_size = parameters.size || DEFAULT_CIRCLE_SIZE;
-        this.m_color = new THREE.Color();
+        this.m_color = getOptionValue(color, new THREE.Color());
+        this.m_opacity = getOptionValue(opacity, 1.0);
 
         this.uniforms = {
+            size: new THREE.Uniform(getOptionValue(size, CirclePointsMaterial.DEFAULT_CIRCLE_SIZE)),
             diffuse: new THREE.Uniform(this.m_color),
-            size: new THREE.Uniform(this.m_size)
+            opacity: new THREE.Uniform(this.m_opacity)
         };
```

Comment:
I know Marcel it is not your code, but we deal with quite gentle setup here, uniforms of material are re-assigned - changed and there is not information for WebGL about it.... Normally in such case you need to call:
```
this.uniformsNeedsUpdate = true;
```
but here you can make it even better, by passing uniforms already in `ShaderMaterial` params to super c-tor. It will be better performance wise... Please take care about it.

Reply:
Ok i will change the implemenation. But I don't think it is really possible to run into a problem. The material is only compiled when it is used during rendering. The uniforms are changed in the constructor of the material so I don't see a chance that it will be compiled before the constructor code has completed.